Update SendGrid cohort export docs (#500)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/destinations/sendgrid-cohort.md

## Considerations

- The API key from SendGrid must be either Full Access or Restricted Access with Marketing permissions.
- Cohorts cannot have duplicate emails, as it will cause sync errors with SendGrid.
  - The recommended way to guarantee unique email addresses is by setting your Amplitude User IDs to use email addresses. 
- All emails that are being exported must be valid; otherwise, the sync will fail.
- You must map the email, first name, and last name fields. If the Amplitude user doesn't have values for the first and last name fields, the corresponding SendGrid contact won't have first or last names, only the email address.
- In cases where users don't have an email address, they aren't synced to SendGrid. This can cause a discrepancy between the number of users in SendGrid and Amplitude. 

> - Emails on SendGrid's side is **not** case-sensitive. Both User@Company.com and user@company.com will sync to the same contact (user@company.com)

Reviewer: ```suggestion
- In SendGrid, emails aren't case-sensitive. For example, "User@Company.com" and "user@company.com" sync to the same contact (user@company.com).
```

> - New contacts will be created for Amplitude users that are not SendGrid contacts, existing contacts will not be duplicated. Both types of users will be added to the list.

Reviewer: ```suggestion
- New contacts are created for Amplitude users that are not SendGrid contacts, and existing contacts aren't duplicated. Both types of users are added to the list.
```
